### DISCUSSION TITLE
How to prioritise deferred tasks that has been completed

### DISCUSSION THREAD

**Discussion by thetkoseek** [apache/airflow#37822] [General]
We use airflow on eks, and spin up 100s of podoperators. We are using deferred podoperators and keep our celery workers light weight. I am not clear on whether we are able to prioritise tasks that are complete from the deferred task. The reason is that if these pods are not cleaned up first, then the resource allocation of the completed pods never clear up. As more complete tasks/pods are waiting to be terminated, the build up of resources continue to climb. Note that these are most likely related to xcom enabled podoperators as the sidecar remain operational until terminated.

My question is whether there is anyway for completed tasks to be picked up by workers first before execution of new tasks?

Thanks for your help.

**Comment by potiuk** [comment]
You have not written which version of Airflow you are at. But that resembles a problem that was fixed in one of the previous versions - you can look it up in the changelog on Airflow website, but generally a solution (if that's it) is to upgrade to latest version that contains a lot of fixes to a lot of issues with every single version.

**Reply by thetkoseek** [reply]
hmm thanks potiuk. I am on version 2.8.1. There is only one newer version, however I cannot find in the changelog that resolves this.

**Reply by potiuk** [reply]
Then I suggest to open an issue with all the details - showing what happens.

**Reply by potiuk** [reply]
(something that will make it possible to reproduce it).
